DOC: concise dependency heading + small clarifications (#27058)
matplotlib/matplotlib: doc/devel/dependencies.rst

.. _dependencies:

************
Dependencies
************

> Runtime

Reviewer: I like the longer versions better because they are more self-contained.
Whenever I read these single-adjective headings, I feel they are incomplete and I should mentally add “dependencies “. Headings should describe the topic of the section. A single adjective cannot do that, because adjectives only qualify things, but that thing is not stated.

Reviewer 2: Self-contained subheadings also work better if you want to link a specific subheading from some other part of the docs I think.

Reviewer 3: I don't feel very strongly, but personally prefer the proposed concise headings.  I think that if it says 

- Dependencies
  - Runtime
  - Testing
  - Documentation
  - etc

the "dependencies" is understood, and is better than 

- Dependencies
  - Runtime dependencies
  - Testing dependencies
  - Documentation dependencies
  - etc dependencies

I appreciate the cross linking point, but we can always inject the proper description``` :ref:`runtime dependencies <runtime_dependencies>` ```

  git clean -xfd

> Install from source

Reviewer: Not quite sure whether "install" is the right term here. But I don't have an alternative either.

Reviewer: I don't want to over-think this, but "source" is a quite generic term. IMHO good headings would be
- C libraries
  - Automatic download (*this section header is missing but should be added*)
  - Use system libraries
  - From source files (or "Custom source files")

.. _development-dependencies:

> C-extensions
> ============

Reviewer 4: They're mostly C++ anyway, but easier to be less specific:
```suggestion
Compiled extensions
===================
```

.. _development-dependencies:

Build dependencies
==================

.. _setup-dependencies:

Python
------

By default, ``pip`` will build packages using build isolation, and the following
dependencies will be automatically installed in the isolated environment to build
Matplotlib. However, for development, you may wish to make an editable install, which
will require disabling build isolation, so these build dependencies should be installed
in the target environment manually.

> :

Reviewer 4: Not sure why this colon is here?

  Also a runtime dependency for editable installs.
- `NumPy <https://numpy.org>`_ (>= 1.21).  Also a runtime dependency.

.. note::

  The Python packages listed here will be installed automatically if you are using
  a virtual environment that was created following the instructions in :ref:`dev-environment`.

.. _compile-dependencies:

> C-extensions
> ------------

Reviewer 4: ```suggestion
Compiled extensions
-------------------
```

Matplotlib requires a C++ compiler that supports C++17, and each platform has a
development environment that must be installed before a compiler can be installed.

> You may also need to install headers for various libraries called in the C-extension
> source files.

Reviewer 4: ```suggestion
You may also need to install headers for various libraries used in the compiled extension
source files.
```

.. include:: ../../requirements/doc/doc-requirements.txt
   :literal:

External tools
--------------

> The documentation will not build without LaTeX and Graphviz.  These are not

Reviewer: Simpler without double negation:
```suggestion
The documentation build requires LaTeX and Graphviz.  These are not
```
